app/components/ExternalLinksBlock/components/ExternalLink/ExternalLink.jsx

```diff
@@ -30,6 +30,11 @@ export default function ExternalLink({ className, isExternalLink, isSocialLink,
     iconClasses = `fa ${icon} fa-fw`;
     linkLabel = path;
     linkProps['aria-label'] = socialIcons[site].ariaLabel;
+    if (site === 'wordpress') {
+      linkProps.href = `https://${path}.${site}`;
+    } else {
+      linkProps.href = `https://${site}${path}`;
+    }
```

Comment:
I'm trying to think of scenarios where `site` or `path` can be compromised, but I can't think of any values that might compromise the website's security.

Reply:
Wait, a thought: up at lines 29 32, there's `const icon = socialIcons[site].icon;` and `linkProps['aria-label'] = socialIcons[site].ariaLabel`

If I try to hack the Social links by setting `site` to an invalid Social media site (say `'twiiitter.com/'`), the code will crash because `socialIcons['twiiitter.com/']` will be `undefined`.

_We can use this._

We just add a new line at 28 that goes something like...
```
if (isSocialLink) {
  // Check whitelist for valid Social Media websites.
  if (!socialIcons[site]) return null;  // Wait a minute we don't recognise your so-called website!
  ...
```